support/Tests/GraphQLSupportTests/Generated/Entry.swift

```diff
@@ -0,0 +1,152 @@
+// Generated from graphql_swift_gen gem
+import Foundation
+
+import GraphQLSupport
+
+public protocol Entry {
+	var typeName: String { get }
+
+	var key: String { get }
+
+	var ttl: Date? { get }
+
+	func childResponseObjectMap() -> [GraphQL.AbstractResponse]
+
+	func responseObject() -> GraphQL.AbstractResponse
+}
+
+extension Generated {
+	open class EntryQuery: GraphQL.AbstractQuery, GraphQLQuery {
+		public typealias Response = Entry
+
+		@discardableResult
+		open func key(aliasSuffix: String? = nil) -> EntryQuery {
+			addField(field: "key", aliasSuffix: aliasSuffix)
+			return self
+		}
+
+		@discardableResult
+		open func ttl(aliasSuffix: String? = nil) -> EntryQuery {
+			addField(field: "ttl", aliasSuffix: aliasSuffix)
+			return self
+		}
+
+		override init() {
+			super.init()
+			addField(field: "__typename")
+		}
+
+		@discardableResult
+		open func onIntegerEntry(subfields: (IntegerEntryQuery) -> Void) -> EntryQuery {
+			let subquery = IntegerEntryQuery()
+			subfields(subquery)
+			addInlineFragment(on: "IntegerEntry", subfields: subquery)
+			return self
+		}
+
+		@discardableResult
+		open func onStringEntry(subfields: (StringEntryQuery) -> Void) -> EntryQuery {
+			let subquery = StringEntryQuery()
+			subfields(subquery)
+			addInlineFragment(on: "StringEntry", subfields: subquery)
+			return self
+		}
+	}
+
+	open class UnknownEntry: GraphQL.AbstractResponse, GraphQLObject, Entry {
+		public typealias Query = EntryQuery
+
+		open override func deserializeValue(fieldName: String, value: Any) throws -> Any? {
+			let fieldValue = value
+			switch fieldName {
+				case "key":
+				guard let value = value as? String else {
+					throw SchemaViolationError(type: UnknownEntry.self, field: fieldName, value: fieldValue)
+				}
+				return value
+
+				case "ttl":
+				if value is NSNull { return nil }
+				guard let value = value as? String else {
+					throw SchemaViolationError(type: UnknownEntry.self, field: fieldName, value: fieldValue)
+				}
+				return iso8601DateParser.date(from: value)!
```

Comment:
🚨 If the date string is in an unexpected format, this will cause our clients to die at runtime with a fatal error that is out of their control.

Instead, the date parsing could be rolled into the `guard` clause so that failure to parse becomes a SchemaViolationError as well.

Reply:
The code to serialize and deserialize a date string is specified to the generator, so this would only need to be improved for the example.  It would be worth checking to see what we do for our shopify ios app though.

Reply:
Looks like we use essentially the same code:
```
return GraphQL.iso8601DateParser.date(from: value)!
```